EVE.pm

```diff
@@ -0,0 +1,210 @@
+
+# Remove temporary files
+rm header variants
+
+# Compress and index
+bgzip ${OUTPUT_FOLDER}/${OUTPUT_NAME};
+
+# If not installed, use: sudo apt install tabix
+tabix ${OUTPUT_FOLDER}/${OUTPUT_NAME}.gz;
+
+### END
+
+=cut
+package EVE;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Variation::Utils::Sequence qw(get_matched_variant_alleles);
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+sub new {
+  my $class = shift;
+
+  my $self = $class->SUPER::new(@_);
+
+  $self->expand_left(0);
+  $self->expand_right(0);
+
+  my $assembly = $self->{config}->{assembly};
+
+  die "\nAssembly is not GRCh38, EVE only works with GRCh38. \n" if ($assembly ne "GRCh38");
+
+  my $param_hash = $self->params_to_hash();
+
+  die "\nERROR: No EVE file specified\nTry using 'file=<path-to>/eve_file.vcf.gz'\n" unless defined($param_hash->{file});
+
+  $self->add_file($param_hash->{file});
+
+  my @valid_class_numbers = (10, 20, 25, 30, 40, 50, 60, 70, 75, 80, 90);
+
+  if (defined($param_hash->{class_number})) {
+
+    die "\nERROR: This class_number: '$param_hash->{class_number}' does not exists.\nTry any of this numbers: " . join(', ', @valid_class_numbers)
+    unless ($param_hash->{class_number} ~~ @valid_class_numbers);
+    $self->{class_number} = $param_hash->{class_number};
+  } else {
+    $self->{class_number} = 75;
+  }
+
+  return $self;
+}
+
+sub feature_types {
+  return ['Transcript'];
+}
+
+sub get_header_info {
+  my $self = shift;
+  return {
+    EVE_SCORE => 'Score from EVE model',
+    EVE_CLASS   => 'Classification (Benign, Uncertain, or Pathogenic) when setting class_number (default 75%) as uncertain'
+  }
+}
+
+sub run {
+  my ($self, $tva) = @_;
+  my $vf = $tva->variation_feature;
+
+  return {} unless grep {$_->SO_term eq 'missense_variant'} @{$tva->get_all_OverlapConsequences};
+
+  # get allele
+  my $alt_allele = $tva->variation_feature_seq;
+  my $ref_allele = $vf->ref_allele_string;
+
+  return {} unless $alt_allele =~ /^[ACGT-]+$/;
+
+  my @data = @{
+    $self->get_data(
+      $vf->{chr},
+      $vf->{start} - 2,
+      $vf->{end}
+    )
+  };
+
+  return {} unless(@data);
+
+  foreach my $variant (@data) {
+
+    my $matches = get_matched_variant_alleles(
+      {
+        ref    => $ref_allele,
+        alts   => [$alt_allele],
+        pos    => $vf->{start},
+        strand => $vf->strand
+      },
+      {
+       ref  => $variant->{ref},
+       alts => [$variant->{alt}],
+       pos  => $variant->{start},
+      }
+    );
+
+    return $variant->{result} if (@$matches);
+  }
+
+  return {};
+
+}
+
+sub parse_data {
+  my ($self, $line) = @_;
+
+  # Parsing VCF fields
+  my ($chrom, $pos, $id, $ref, $alt, $qual, $filter, $info) = split /\t/, $line; 
+
+  # Parsing INFO field
+  my ($EVE_SCORE) = $info =~ /EVE=(.*?);/;
+  my $class_number = $self->{class_number};
+  my ($EVE_CLASS) = $info =~ /Class$class_number=(.*?);/;
```

Comment:
This pattern match requires a ';' which we don't have for the final value, so requesting class_number=90 returns no EVE_CLASS.

Reply:
Hi @sarahhunt, fixed regex bug and updated header to include `class_number`.